xbmc/powermanagement/IPowerSyscall.h

```diff
@@ -64,6 +64,12 @@ class IPowerSyscall
   virtual bool PumpPowerEvents(IPowerEventsCallback *callback) = 0;
 
   static const int MAX_COUNT_POWER_FEATURES = 4;
+
+  /**\brief Called by power manager to create platform power system adapter
+  *
+  * This method used to create platfrom specified power system adapter
+  */
+  static IPowerSyscall* CreateInstance();
```

Comment:
why not having platforms register a create function? as far as I can see this is an optional component.

Reply:
@FernetMenta I've updated this, but I'm not sure that I did this right.

Reply:
@afedchin I would drop CNullPowerSyscall and check m_instance instead

Reply:
@FernetMenta done. please check. is windowing right place to register those things?

Reply:
@afedchin should do for now. Maybe we find a better place later.